src/attr/_make.py

```diff
@@ -912,20 +912,19 @@ def wrap(cls):
             raise TypeError(
                 "Invalid value for hash.  Must be True, False, or None."
             )
-        elif hash is False or (hash is None and cmp is False):
+        elif hash is False or (hash is None and cmp is False) or is_exc:
+            # Hash by ID aka NOP.
```

Comment:
A minor quibble about the comment: since this leaves `__hash__` untouched, wouldn't the behavior here be "Use any `__hash__` provided by the user; otherwise fall back to using the `__hash__` of the base class.  If the base class is `object`, this is hashing by ID."?  The case of a user-supplied hash is slightly unclear in [the documentation](https://github.com/python-attrs/attrs/blob/8174b03b962165278dd65508bf0f947c98a1190a/src/attr/_make.py#L773) as well.